install/test/helm_test.go

```diff
@@ -6293,12 +6309,126 @@ metadata:
+					Entry("29-knative-internal-proxy-deployment.yaml", "knative-internal-proxy", "knative-internal-proxy", "Deployment", "settings.integrations.knative.proxy.containerSecurityContext", "settings.integrations.knative.version=0.8.0", "settings.integrations.knative.enabled=true"),
+					Entry("19-gloo-mtls-certgen-cronjob.yaml", "gloo-mtls-certgen-cronjob", "certgen", "CronJob", "gateway.certGenJob.containerSecurityContext", "global.glooMtls.enabled=true", "gateway.certGenJob.cron.enabled=true"),
+				)
+
+				// Helper function to apply the default restricted container security context
+				type applyContainerSecurityDefaults func(*corev1.SecurityContext)
+
+				applyDiscovery := applyContainerSecurityDefaults(func(securityContext *corev1.SecurityContext) {
+					securityContext.ReadOnlyRootFilesystem = pointer.Bool(true)
+					securityContext.RunAsUser = pointer.Int64(int64(10101))
+				})
+				applyNil := applyContainerSecurityDefaults(func(securityContext *corev1.SecurityContext) {})
+				applyRunAsUser := applyContainerSecurityDefaults(func(securityContext *corev1.SecurityContext) {
+					securityContext.RunAsUser = pointer.Int64(int64(10101))
+				})
+				applyKnative := applyContainerSecurityDefaults(func(securityContext *corev1.SecurityContext) {
+					securityContext.RunAsUser = pointer.Int64(int64(10101))
+					securityContext.ReadOnlyRootFilesystem = pointer.Bool(true)
+					securityContext.Capabilities = &corev1.Capabilities{
+						Drop: []corev1.Capability{"ALL"},
+						Add:  []corev1.Capability{"NET_BIND_SERVICE"},
+					}
+				})
+				applyClusterIngress := applyContainerSecurityDefaults(func(securityContext *corev1.SecurityContext) {
+					securityContext.Capabilities = &corev1.Capabilities{
+						Drop: []corev1.Capability{"ALL"},
+						Add:  []corev1.Capability{"NET_BIND_SERVICE"},
+					}
+					securityContext.ReadOnlyRootFilesystem = pointer.Bool(true)
+				})
+
+				getDefaultRestrictedContainerSecurityContext := func(seccompType string, applyContainerDefaults applyContainerSecurityDefaults) *corev1.SecurityContext {
+					// Use default value if not set
+					if seccompType == "" {
+						seccompType = "RuntimeDefault"
+					}
+
+					defaultRestrictedContainerSecurityContext := &corev1.SecurityContext{
+						RunAsNonRoot:             pointer.Bool(true),
+						AllowPrivilegeEscalation: pointer.Bool(false),
+						Capabilities: &corev1.Capabilities{
+							Drop: []corev1.Capability{"ALL"},
+						},
+						SeccompProfile: &corev1.SeccompProfile{
+							Type: corev1.SeccompProfileType(seccompType),
+						},
+					}
+					applyContainerDefaults(defaultRestrictedContainerSecurityContext)
+					return defaultRestrictedContainerSecurityContext
+				}
+
+				DescribeTable("applies default retricted container security contexts", func(resourceName string, containerName string, resourceType string, securityRoot string, applyDefaults applyContainerSecurityDefaults, extraArgs ...string) {
```

Comment:
```suggestion
				DescribeTable("applies default retricted container security contexts", func(resourceName string, containerName string, resourceType string, securityRoot string, applyDefaults applyContainerSecurityDefaults, extraArgs ...string) {
```

Also, the `securityRoot` arg doesn't appear to be used

Reply:
Can't apply the suggestion, but removed securityContext from the args and Entries

Reply:
Opened another thread about the typo that I meant to point out with the suggestion